src/main/java/ladder/domain/Line.java

```diff
@@ -0,0 +1,135 @@
+package ladder.domain;
+
+import ladder.pattern.ValueGenerateStrategy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static ladder.domain.Point.FIRST_POSITION;
+import static ladder.domain.Point.UNIT_HORIZONTAL_LINE_LENGTH;
+
+public class Line {
+    private static final int MIN_WIDTH = 2;
+
+    private final List<Point> points;
+
+    public Line(List<Point> points) {
+        validatePoints(points);
+        this.points = points;
+    }
+
+    public static Line create(int width, ValueGenerateStrategy valueGenerateStrategy) {
+        validateWidth(width);
+        List<Point> points = new ArrayList<>();
+        Point point = Point.first(valueGenerateStrategy.booleanGenerate());
+        points.add(point);
+        for (int index = FIRST_POSITION + UNIT_HORIZONTAL_LINE_LENGTH; index < width - UNIT_HORIZONTAL_LINE_LENGTH; index++) {
+            point = point.next(valueGenerateStrategy.booleanGenerate());
+            points.add(point);
+        }
+        points.add(point.last());
+        return new Line(points);
+    }
+
+    public List<Point> points() {
+        return points;
+    }
+
+    public void validatePoints(List<Point> points) {
+        if (points == null) {
+            throw new IllegalArgumentException("라인을 구성하는 포인트들은 null 일 수 없습니다.");
+        }
+        validateWidth(points.size());
+        validateDirectionAndPosition(points);
+    }
+
+    private void validateDirectionAndPosition(List<Point> points) {
+        int width = points.size();
+        Point firstPoint = points.get(FIRST_POSITION);
+        validateFirstPoint(firstPoint);
+
+        Point previousPoint = firstPoint;
+        for (int index = UNIT_HORIZONTAL_LINE_LENGTH; index < width - UNIT_HORIZONTAL_LINE_LENGTH; index++) {
+            Point currentPoint = points.get(index);
+            validateMiddlePoint(previousPoint, currentPoint);
+            previousPoint = currentPoint;
+        }
+
+        Point lastPoint = points.get(width - UNIT_HORIZONTAL_LINE_LENGTH);
+        validateLastPoint(lastPoint, width);
+    }
+
+    private void validateFirstPoint(Point firstPoint) {
+        validateFirstPointDirection(firstPoint);
+        validateFirstPointPosition(firstPoint);
```

Comment:
```suggestion
        if (!firstPoint.isCorrectFirstPosition()) {
            throw new IllegalArgumentException("~~에러메시지~~");
        }
```
Point객체에게 올바른 첫 번째 포지션인지에 대한 검증을 위임하면 Line에서 검증 코드에 대한 복잡도를 낮출 수 있지 않을까요?

middlePoint, lastPoint 의 검증도 가능할까요?